lxd/seccomp/seccomp.go

```diff
@@ -1675,6 +1692,110 @@ func (s *Server) HandleSchedSetschedulerSyscall(c Instance, siov *Iovec) int {
 	return 0
 }
 
+// HandleSysinfoSyscall handles sysinfo syscalls.
+func (s *Server) HandleSysinfoSyscall(c Instance, siov *Iovec) int {
+	l := logger.AddContext(logger.Log, logger.Ctx{
+		"container":             c.Name(),
+		"project":               c.Project(),
+		"syscall_number":        siov.req.data.nr,
+		"audit_architecture":    siov.req.data.arch,
+		"seccomp_notify_id":     siov.req.id,
+		"seccomp_notify_flags":  siov.req.flags,
+		"seccomp_notify_pid":    siov.req.pid,
+		"seccomp_notify_fd":     siov.notifyFd,
+		"seccomp_notify_mem_fd": siov.memFd,
+	})
+
+	defer l.Debug("Handling sysinfo syscall")
+
+	runtime.LockOSThread()
```

Comment:
I think you want an immediate `runtime.UnlockOSThread()` here? I'm not sure what happens if you return from the go routine but not explicitly yield the lock.

Reply:
I've removed the lock (at least for now) as I'm having trouble with accessing the cgroup usage info (via `c.Cgroup()`) as it seems to be just hanging. I don't know why as these same calls (e.g. `cg.GetMemoryLimit()`) are working fine (AFAIK) when call from the metrics consumer endpoint. My next step is to confirm this assumption before I continue any further.